Commit 18ad1c654448 "[net] introduce csum_replace_by_diff() helper" (upstream
8050c0f0274a "bpf: allow bpf_csum_diff to feed bpf_l3_csum_replace as well")
introduced csum_replace_by_diff() helper function, which is subsequently
utilised in commit c3e00dfce489 "[netdrv] i40e/i40evf: Fix TSO checksum
pseudo-header adjustment" (upstream 24d41e5e2c9a).

Backport change: add implementation of csum_replace_by_diff() from "[net]
introduce csum_replace_by_diff() helper" to i40e_backport_compat.h.

Index: src/drivers/net/ethernet/intel/i40evf/i40e_backport_compat.h
===================================================================
--- src.orig/drivers/net/ethernet/intel/i40evf/i40e_backport_compat.h	2017-04-27 20:21:23.706222692 +0200
+++ src/drivers/net/ethernet/intel/i40evf/i40e_backport_compat.h	2017-04-27 20:22:08.913686840 +0200
@@ -1,9 +1,17 @@
 #ifndef _RH_I40EVF_BACKPORT_COMPAT_H_
 #define _RH_I40EVF_BACKPORT_COMPAT_H_
 
+#include <net/checksum.h>
+
+
 #define NETIF_F_TSO_MANGLEID 0
 #define NETIF_F_GSO_PARTIAL  0
 #define SKB_GSO_PARTIAL      (1 << 16)
 
 
+static inline void csum_replace_by_diff(__sum16 *sum, __wsum diff)
+{
+	*sum = csum_fold(csum_add(diff, ~csum_unfold(*sum)));
+}
+
 #endif /* #ifndef _RH_I40EVF_BACKPORT_COMPAT_H_ */
